src/hotspot/share/utilities/nonblockingQueue.inline.hpp

```diff
@@ -116,7 +116,10 @@ void NonblockingQueue<T, next_ptr>::append(T& first, T& last) {
     // Successfully extended the queue list from old_tail to first.  No
     // other push/append could have competed with us, because we claimed
     // old_tail for extension.  We won any races with try_pop by changing
-    // away from end-marker.  So we're done.
+    // away from end-marker.  So we're done.  Note that ABA is possible; a
+    // concurrent try_pop could take old_tail before our update, it gets
+    // recycled and re-added to the end, and then we successfully cmpxchg,
+    // rendering the list in _tail circular.
```

Comment:
Suggestions:

1. start the new comment on a new line
2.  "Note that ABA would be possible if a concurrent try_pop takes old_tail before our update, ...

Cheers,
David

Reply:
OK, I separated the note and reworded it a bit.  Better?